### ISSUE TITLE
Black screen when using IStyleable.StyleKey on Window

### ISSUE THREAD

**Issue by ydogus** [AvaloniaUI/Avalonia#1258]
Hi,
I've a very basic scenario:
Defined a new `FooWindow` class based on `Window`. I also implemented `IStyleable` and defined `IStyleable.StyleKey` explicitly. That's all. When I run the app I got this black window.
![image](https://user-images.githubusercontent.com/3532488/32359174-3b6b5770-c04c-11e7-9f88-a1238ebc4a6f.png)

I doubt this is the expected behavior. :)
When I comment `StyleKey` property the problem goes away.

The project is very minimal. Here are the files if you want to reproduct it:
App.xaml:
```xaml
<Application xmlns="https://github.com/avaloniaui">
    <Application.Styles>
        <StyleInclude Source="resm:Avalonia.Themes.Default.DefaultTheme.xaml?assembly=Avalonia.Themes.Default"/>
    </Application.Styles>
</Application>
```
App.xaml.cs:
```csharp
public class App : Application
{
    public override void Initialize()
    {
        AvaloniaXamlLoader.Load(this);
        base.Initialize();
    }

    static void Main(string[] args)
    {
        var builder = AppBuilder.Configure<App>();
        builder.UsePlatformDetect();
        builder.Start<FooWindow>();
    }
}
```
FooWindow.cs:
```csharp
public class FooWindow : Window, IStyleable
{
    Type IStyleable.StyleKey => typeof(FooWindow);
}
```

I'm using only Avalonia and Avalonia.Desktop packages.
* Avalonia Version: `0.5.2-build4199-alpha`.
* .NET Core 2.0
* Windows 10 (1703)
* Also tried it on Fedora 26 and got the same result.

**Comment by jkoritzinsky** [comment]
If you redefine the style key a type, you need to specify a new style for that type. In this case, there is no style, and as a result no `ControlTemplate` for the Window. If you look at our inheritance hierarchy, `Window` inherits from `ContentControl`, which needs a template with a `ContentPresenter` element with the name `PART_ContentPresenter` in it. Without that `ContentPresenter`, the `ContentControl` (in this case a `FooWindow`) will not have any visual children since the `FooWindow` doesn't have a `ContentPresenter` in its template. Since it has no visual children, our rendering system doesn't see anything to render (by design). When the rendering system doesn't render anything, you get a black window, as you see.


In summary, this is expected behavior. When you redefine the `IStyleable.StyleKey` for a subclass of `Window` you need to create a style with a `ControlTemplate` that has a `ContentPresenter` named `PART_ContentPresenter` in it, otherwise nothing will render and you will get a black screen.
